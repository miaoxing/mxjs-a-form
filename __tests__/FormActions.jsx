import FormActions from '../FormActions';
import {MemoryRouter} from 'react-router';
import {render} from '@testing-library/react';
import {ConfigProvider} from 'antd';

describe('FormAction', () => {
  test('render', async () => {
    const {container} = render(
      <ConfigProvider theme={{hashed: false}}>
        <MemoryRouter>
          <FormActions/>
        </MemoryRouter>
      </ConfigProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  test('custom offset', async () => {
    const {container} = render(
      <ConfigProvider theme={{hashed: false}}>
        <MemoryRouter>
          <FormActions wrapperCol={{offset: 8}}/>
        </MemoryRouter>
      </ConfigProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
