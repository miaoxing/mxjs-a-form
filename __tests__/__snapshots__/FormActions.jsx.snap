// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAction custom offset 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered mb-6 [&_.ant-card-body>:last-child]:mb-0"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-offset-8 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                >
                  <div
                    class="ant-space-item"
                  >
                    <button
                      class="ant-btn ant-btn-primary"
                      type="submit"
                    >
                      <span>
                        提 交
                      </span>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <a
                      class="ant-btn ant-btn-default"
                      href="/"
                      tabindex="0"
                    >
                      <span>
                        返回列表
                      </span>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FormAction render 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered mb-6 [&_.ant-card-body>:last-child]:mb-0"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-8 ant-col-offset-4 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                >
                  <div
                    class="ant-space-item"
                  >
                    <button
                      class="ant-btn ant-btn-primary"
                      type="submit"
                    >
                      <span>
                        提 交
                      </span>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <a
                      class="ant-btn ant-btn-default"
                      href="/"
                      tabindex="0"
                    >
                      <span>
                        返回列表
                      </span>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
